src/opera/threading/utils.py

```diff
@@ -1,5 +1,18 @@
-from threading import current_thread  # type: ignore # pylint: disable=no-name-in-module
+from io import TextIOWrapper
+from threading import current_thread, Lock  # type: ignore # pylint: disable=no-name-in-module
 
 
 def print_thread(string):
     print(f"[{current_thread().name}] {string}")
+
+
+class SafePrinter:
+    lock = Lock()
+
+    @classmethod
+    def print_lines(cls, wrapper: TextIOWrapper):
+        cls.lock.acquire()
+        print_thread("------------")
+        for line in wrapper:
+            print(line.rstrip())
+        cls.lock.release()
```

Comment:
Sanity tests fail with `opera/threading/utils.py|14:8|SafePrinter.print_lines |R1732|consider-using-with|Consider using 'with' for resource-allocating operations`, so you could probably change these lines to:

```suggestion
        with cls.lock:
            print_thread("------------")
            for line in wrapper:
                print(line.rstrip())
```

Reply:
Done.